rgw/v2/tests/s3_swift/reusable.py

```diff
@@ -595,6 +596,124 @@ def upload_mutipart_object(
         log.info("multipart upload complete for key: %s" % s3_object_name)
 
 
+def upload_part(
+    rgw_client,
+    s3_object_name,
+    bucket_name,
+    mpu,
+    part_number,
+    body,
+    content_length,
+    parts_info,
+):
+    try:
+        part_upload_response = rgw_client.upload_part(
+            Bucket=bucket_name,
+            Key=s3_object_name,
+            PartNumber=part_number,
+            UploadId=mpu["UploadId"],
+            Body=body,
+            ContentLength=content_length,
+        )
+        log.info(f"part uploaded response {part_upload_response}")
+    except Exception as e:
+        log.info(e)
+        return
+    part_info = {"PartNumber": part_number, "ETag": part_upload_response["ETag"]}
+    parts_info["Parts"].append(part_info)
+
+
+def test_multipart_upload_failed_parts(
+    rgw_client, s3_object_name, bucket_name, part1_path, part2_path
+):
+    parts_info = {"Parts": []}
+    log.info(f"no of parts: 2")
```

Comment:
@hmaheswa, do we need format string here?

Reply:
@anrao19 
done, please review again